### PR TITLE
Model change/payment-invoice-receipt-status-change

### PR DESCRIPTION
**Pull request by rhimmelbauer**
Notes:

Changes in how status are for Model Receipt, Payment and Invoice. The new logic goes as follows:

Invoice
An Invoice status can be Cart, Checkout or Complete.

Payment:
A payment status can be Queued, Active, Authorized, Caputred, Complete, Canceled, or Refunded.

Receipt:
A receipt has no longer a status as it represents a successfull transaction

-  merged develop to working branch 
-  added check for get expired cards if none return 
-  removed PurchaseStatus from receipt 
-  updated unit tests 
-  added migrations to update InvoiceStatus, PurchaseStatus from models Invoice, Receipt and Payment 
-  moved PurchaseStatus to payment from receipt 
-  changed InvoiceStatus calls to only use Cart, Checkout or Complete 
-  moved InvoiceStatus outside of the model and updated reference in files 

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

File: vendor/models/payment.py [hunk 1]
```diff
@@ -32,6 +33,7 @@ class Payment(SoftDeleteModelBase):
     success = models.BooleanField(_("Successful"), default=False)
     payee_full_name = models.CharField(_("Name on Card"), max_length=50)
     payee_company = models.CharField(_("Company"), max_length=50, blank=True, null=True)
+    status = models.IntegerField(_("Status"), choices=PurchaseStatus.choices, default=0)
```
Comment (mgcamisa) [inline vendor/models/payment.py:36]: PurchaseStatus doesn't appear to have a choice with the value 0- maybe this default should be 1?

File: vendor/processors/authorizenet.py [hunk 2]
```diff
@@ -656,7 +657,7 @@ def get_customer_id_for_expiring_cards(self, month):
         customer_profile_ids = []
         last_page = 1
         
-        if self.transaction_submitted:
+        if self.transaction_submitted and response.paymentProfiles.paymentProfile:
```
Comment (mgcamisa) [inline vendor/processors/authorizenet.py:660]: thank you!

File: vendor/tests/test_processor.py [hunk 3]
```diff
@@ -1083,10 +1083,18 @@ def test_get_expiring_cards_success(self):
...
+        ids = processor.get_customer_id_for_expiring_cards("2024-01")
 
         self.assertTrue(processor.transaction_submitted)
+        self.assertTrue(ids)
 
+    def test_get_expiring_cards_empty(self):
+        site = Site.objects.get(pk=1)
+        processor = AuthorizeNetProcessor(site)
+
+        ids = processor.get_customer_id_for_expiring_cards("2020-01")
```
Comment (mgcamisa) [inline vendor/tests/test_processor.py:1095]: do you want to add a `processor.transaction_sumbitted` check here too?

File: vendor/models/payment.py (hunk 1 above)
Comment (rhimmelbauer) [reply vendor/models/payment.py:36]: No it is on purpose :)

File: vendor/models/receipt.py [hunk 4]
```diff
@@ -47,11 +46,11 @@ def void(self):
         self.meta['voided_on'] = dateformat.format(self.end_date, 'Y-M-d H:i:s')
 
     def cancel(self):
-        self.status = PurchaseStatus.CANCELED
```
[the remark this answers is not in the document]
Comment (rhimmelbauer) [reply vendor/models/receipt.py:50]: yes, that is coming up in the another pr I am working on

File: vendor/migrations/0036_auto_20220602_2233.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.2.13 on 2022-06-02 22:33
```
[the remark this answers is not in the document]
Comment (rhimmelbauer) [reply vendor/migrations/0036_auto_20220602_2233.py:1]: No, when dealing with deleted fields and merging data it is better to be very explicit

**Pull request merged**